Port credentials dialogs to React

KKoukiou opened the pull request:
main:
![image](https://user-images.githubusercontent.com/200109/137462786-92618d6a-24b9-4a21-9435-520a196d8330.png)

this PR:
![image](https://user-images.githubusercontent.com/200109/137703305-6760d8a9-1203-436b-b2d4-138a2a173f00.png)

martinpitt commented:
[this failure](https://logs.cockpit-project.org/logs/pull-16470-20211014-154638-19ae429e-fedora-coreos/log.html#294) looks relevant? Supposedly coreos is a special snowflake here as it always authenticates through ssh?

KKoukiou commented:
> Thanks! This generally works very well now. I updated the screenshot in the description. The only thing that's broken now is the "x" close button in the modal, it's apparently not wired up? That should be simple to fix?

Fixed. Thanks for noticing!
